main.go

```diff
@@ -1,93 +1,17 @@
 package main
 
 import (
-	"flag"
-	"time"
+	"fmt"
 
-	"github.com/golang/glog"
-	kubeinformers "k8s.io/client-go/informers"
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/tools/clientcmd"
-	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
-	// _ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
-
-	clientset "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned"
-	informers "github.com/xmudrii/etcdproxy-controller/pkg/client/informers/externalversions"
-	"github.com/xmudrii/etcdproxy-controller/pkg/controller/etcdproxy"
+	"github.com/xmudrii/etcdproxy-controller/pkg/cmd/controller"
 	"github.com/xmudrii/etcdproxy-controller/pkg/signals"
 )
 
-var (
-	kubeconfig string
-	namespace  string
-
-	coreEtcdURL             string
-	coreEtcdCAConfigMapName string
-	coreEtcdCertSecretName  string
-
-	proxyEtcdImage string
-)
-
 func main() {
-	flag.Parse()
-
-	// set up signals so we handle the first shutdown signal gracefully
 	stopCh := signals.SetupSignalHandler()
 
-	cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
-	if err != nil {
-		glog.Fatalf("Error building kubeconfig: %s", err.Error())
-	}
-
-	kubeClient, err := kubernetes.NewForConfig(cfg)
-	if err != nil {
-		glog.Fatalf("Error building kubernetes clientset: %s", err.Error())
-	}
-
-	etcdproxyClient, err := clientset.NewForConfig(cfg)
-	if err != nil {
-		glog.Fatalf("Error building example clientset: %s", err.Error())
-	}
-
-	controllerNamespace, err := etcdproxy.ControllerNamespace(namespace)
-	if err != nil {
-		glog.Fatalf("Error detecting controller namespace: %s", err.Error())
+	cmd := controller.NewCommandEtcdProxyControllerStart(stopCh)
+	if err := cmd.Execute(); err != nil {
+		fmt.Errorf("unable to run etcdproxy command: %v", err)
```

Comment:
This doesn't actually print the error.

`fmt.Fprintf(os.Stderr, message)`

Reply:
Actually, reading it, just `glog.Fatal(err)` will do

Reply:
Fixed.